core/src/test/java/org/apache/james/core/MailAddressTest.java

```diff
@@ -1,241 +1,253 @@
+            assertThat(false).describedAs(e.getMessage()).isTrue();
+        }
+    }
+
+    /**
+     * Test method for {@link MailAddress#getLocalPart()}.
+     */
+    @Test
+    public void testGetLocalPart() {
+
+        try {
+            MailAddress a = new MailAddress(new InternetAddress(GOOD_QUOTED_LOCAL_PART));
+            assertThat(a.getLocalPart().equals(GOOD_LOCAL_PART)).describedAs(GOOD_LOCAL_PART + " != " + a.getLocalPart()).isTrue();
+        } catch (AddressException e) {
+            System.out.println("AddressException" + e.getMessage());
+            assertThat(false).describedAs(e.getMessage()).isTrue();
+        }
+    }
+
+    /**
+     * Test method for {@link MailAddress#toString()}.
+     */
+    @Test
+    public void testToString() {
+
+        try {
+            MailAddress a = new MailAddress(new InternetAddress(GOOD_ADDRESS));
+            assertThat(a.toString().equals(GOOD_ADDRESS)).describedAs(a.toString() + " != " + GOOD_ADDRESS).isTrue();
+        } catch (AddressException e) {
+            System.out.println("AddressException" + e.getMessage());
+            assertThat(false).describedAs(e.getMessage()).isTrue();
+        }
+    }
+
+    /**
+     * Test method for {@link MailAddress#toInternetAddress()}.
+     */
+    @Test
+    public void testToInternetAddress() {
+
+        try {
+            InternetAddress b = new InternetAddress(GOOD_ADDRESS);
+            MailAddress a = new MailAddress(b);
+            assertThat(a.toInternetAddress().equals(b)).isTrue();
+            assertThat(a.toString().equals(GOOD_ADDRESS)).describedAs(a.toString() + " != " + GOOD_ADDRESS).isTrue();
+        } catch (AddressException e) {
+            System.out.println("AddressException" + e.getMessage());
+            assertThat(false).describedAs(e.getMessage()).isTrue();
+        }
+    }
+
+    /**
+     * Test method for {@link MailAddress#equals(java.lang.Object)}.
+     *
+     * @throws AddressException
+     */
+    @Test
+    public void testEqualsObject() throws AddressException {
+
+        MailAddress a = new MailAddress(GOOD_ADDRESS);
+        MailAddress b = new MailAddress(GOOD_ADDRESS);
+
+        assertThat(a.equals(b)).describedAs(a.toString() + " != " + b.toString()).isTrue();
+        assertThat(a.equals(null)).describedAs(a.toString() + " != " + null).isFalse();
+    }
+
+    @Test
+    public void equalsShouldReturnTrueWhenBothNullSender() {
+        assertThat(MailAddress.nullSender())
+            .isEqualTo(MailAddress.nullSender());
+    }
+
+    @Test
+    public void getMailSenderShouldReturnNullSenderWhenNullSender() {
+        assertThat(MailAddress.getMailSender(MailAddress.NULL_SENDER_AS_STRING))
+            .isEqualTo(MailAddress.nullSender());
+    }
+
+    @Test
+    public void getMailSenderShouldReturnParsedAddressWhenNotNullAddress() throws Exception {
+        assertThat(MailAddress.getMailSender(GOOD_ADDRESS))
+            .isEqualTo(new MailAddress(GOOD_ADDRESS));
+    }
+
```

Comment:
please, try to commit white space changes in an other commit (maybe next time)

Reply:
Yes maybe next time. Sorry, I did not even noticed it...